File: Zeze/Arch/ProviderLoad.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Zeze.Arch
{
    public class ProviderLoad : ProviderLoadBase
    {
        public Online Online { get; }

        public ProviderLoad(Online online)
            : base(online.ProviderApp.Zeze)
        {
            Online = online;
        }

        public override LoadConfig GetLoadConfig()
        {
            return Online.ProviderApp.Distribute.LoadConfig;
        }

        public override int GetOnlineLocalCount()
        {
            return Online.LocalCount;
        }

        public override long GetOnlineLoginTimes()
        {
            return Online.LoginTimes;
        }

        public override string GetProviderIp()
        {
            return Online.ProviderApp.DirectIp;
        }

        public override int GetProviderPort()
        {
            return Online.ProviderApp.DirectPort;
        }
    }
}

File: ZezeJava/ZezeJava/src/main/java/Zeze/Arch/ProviderLoad.java
package Zeze.Arch;

public class ProviderLoad extends ProviderLoadBase {
	public final Online online;

	public ProviderLoad(Online online) {
		super(online.providerApp.zeze);
		this.online = online;
	}

	@Override
	public int getOnlineLocalCount() {
		return online.getLocalCount();
	}

	@Override
	public long getOnlineLoginTimes() {
		return online.getLoginTimes();
	}

	@Override
	public LoadConfig getLoadConfig() {
		return online.providerApp.distribute.loadConfig;
	}

	@Override
	public String getProviderIp() {
		return online.providerApp.directIp;
	}

	@Override
	public int getProviderPort() {
		return online.providerApp.directPort;
	}
}
